src/analyze.cpp

```diff
@@ -1442,8 +1449,8 @@ ZigType *analyze_type_expr(CodeGen *g, Scope *scope, AstNode *node) {
 ZigType *get_generic_fn_type(CodeGen *g, FnTypeId *fn_type_id) {
     ZigType *fn_type = new_type_table_entry(ZigTypeIdFn);
     buf_resize(&fn_type->name, 0);
-    const char *cc_str = calling_convention_fn_type_str(fn_type->data.fn.fn_type_id.cc);
-    buf_appendf(&fn_type->name, "%s", cc_str);
+    if (fn_type->data.fn.fn_type_id.cc == CallingConventionC)
+        buf_append_str(&fn_type->name, "extern ");
```

Comment:
I was thinking we could drop `extern` for the C calling convention, once we have `callconv`. Any reason to keep using `extern` for C calling convention functions?

Reply:
Decoupling `extern` from the CC is definitely a good idea, I tried not to change the status quo too much in this PR so I think I'll leave the breaking changes to you :)